The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/21253666459/job/61162059668) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/61162059668) [(plain)](https://github.com/rust-lang/rust/commit/4a976c5e1d68f962a49906d05d65f586a64f946d/checks/61162059668/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
checking C++ file formatting
spellchecking files
building external tool typos from package typos-cli@1.38.1
finished building tool typos
error: `occuring` should be `occurring`
    ╭▸ library/std/src/sys/thread/mod.rs:145:8
    │
145 │     // occuring on older Windows systems.
    ╰╴       ━━━━━━━━
rerun tidy with `--extra-checks=spellcheck --bless` to fix typos
tidy [extra_checks]: checks with external tool 'typos' failed
tidy [extra_checks]: FAIL
tidy: The following check failed: extra_checks
Command `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy /checkout /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo /checkout/obj/build 4 /node/bin/yarn --extra-checks=py,cpp,js,spellcheck` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1612:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1358:29

Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `test src/tools/tidy tidyselftest --extra-checks=py,cpp,js,spellcheck`
Build completed unsuccessfully in 0:02:36
  local time: Thu Jan 22 15:19:05 UTC 2026
  network time: Thu, 22 Jan 2026 15:19:05 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/library/std/src/sys/thread/mod.rs
+++ b/library/std/src/sys/thread/mod.rs
@@ -139,9 +139,16 @@ cfg_select! {
 pub fn sleep_until(deadline: crate::time::Instant) {
     use crate::time::Instant;
 
-    let now = Instant::now();
-
-    if let Some(delay) = deadline.checked_duration_since(now) {
+    // The clock source used for `sleep` might not be the same used for `Instant`.
+    // Since this function *must not* return before the deadline, we recheck the
+    // time after every call to `sleep`. See #149935 for an example of this
+    // occurring on older Windows systems.
+    while let Some(delay) = deadline.checked_duration_since(Instant::now()) {
+        // Sleep for the estimated time remaining until the deadline.
+        //
+        // If your system has a better way of estimating the delay time or
+        // provides a way to sleep until an absolute time, specialize this
+        // function for your system.
         sleep(delay);
     }
 }

